RedBookPlayer/PlayableDisc.cs

```diff
@@ -0,0 +1,662 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Aaru.CommonTypes.Enums;
+using Aaru.CommonTypes.Structs;
+using Aaru.DiscImages;
+using Aaru.Helpers;
+using static Aaru.Decoders.CD.FullTOC;
+
+namespace RedBookPlayer
+{
+    public class PlayableDisc
+    {
+        #region Public Fields
+
+        /// <summary>
+        /// Indicate if the disc is ready to be used
+        /// </summary>
+        public bool Initialized { get; private set; } = false;
+
+        /// <summary>
+        /// Current track number
+        /// </summary>
+        public int CurrentTrackNumber
+        {
+            get => _currentTrackNumber;
+            set
+            {
+                // Unset image means we can't do anything
+                if(_image == null)
+                    return;
+
+                // Check if we're incrementing or decrementing the track
+                bool increment = value >= _currentTrackNumber;
+
+                // Ensure that the value is valid, wrapping around if necessary
+                if(value >= _image.Tracks.Count)
+                    _currentTrackNumber = 0;
+                else if(value < 0)
+                    _currentTrackNumber = _image.Tracks.Count - 1;
+                else
+                    _currentTrackNumber = value;
+
+                // Cache the current track for easy access
+                Track track = _image.Tracks[CurrentTrackNumber];
+
+                // Set new track-specific data
+                byte[] flagsData = _image.ReadSectorTag(track.TrackSequence, SectorTagType.CdTrackFlags);
```

Comment:
Can be null.
Can be byte[0].
Can throw an exception (Aaru 6 wont).

Reply:
With my below comment, I'm guessing `ReadSectorTag` was being used incorrectly here.

Reply:
I did have this incorrect, though that is being fixed differently than what I expected.

Reply:
You're still accessing flagsData without checking if it is null